File: Mattew3748/2748.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim();
const n = Number(input);

let memo = { 0: 0n, 1: 1n };

function fib(n) {
    if(n in memo) return memo[n]

    if(n <= 1) return n

    return memo[n] = fib(n-1) + fib(n-2)
}

console.log(fib(n).toString());

File: tastingcode/2748.py
memo = [-1] * (90 + 1)
memo[0] = 0
memo[1] = 1

n = int(input())
def f(n):
    if memo[n] != -1:
        return memo[n]

    if memo[n] == -1:
        memo[n] = f(n - 2) + f(n - 1)

    return f(n - 2) + f(n - 1)

print(f(n))